src/mqtt/client.rs

```diff
@@ -73,12 +73,12 @@ pub struct MqttClientConfiguration<'a> {
     pub skip_cert_common_name_check: bool,
     #[cfg(not(esp_idf_version = "4.3"))]
     pub crt_bundle_attach: Option<unsafe extern "C" fn(conf: *mut c_types::c_void) -> esp_err_t>,
+    
+    pub cert_pem: &'a [u8],
+    pub client_cert_pem: &'a [u8],
```

Comment:
Regarding ESP IDF 5: If your changes do not compile under ESP IDF 5 (don't they?) the absolute minimum is to conditionalize your code so that the ESP-IDF-4-only variables are exposed only when `#[cfg(esp_idf_version_major = "4")]`

Reply:
Why don't you just compile it for ESP IDF 5? If it compiles, no need to conditionalize it.